### PR TITLE
feat: Add HMAC-SHA256 signature method for OAuth 1.0

### LINKED ISSUE #702: OAuthHmacSigner only supports HMAC-SHA1 (no ability to use HMAC-SHA256 which is required by NetSuite)
Description:
As a NetSuite integration developer, I would like to be able to use HMAC-SHA256, so that I can connect to Netsuite when HMAC-SHA1 is deprecated after August 2021.

Solution:
I would like to have the ability to sign OAuth 1.0 requests using HMAC-SHA256.

Other Information:
As the signature method is hard-coded in the OAuthHmacSigner class, the only way to support this would be a new signer class, or a code change to make the signature method configurable.

Additional Information:
NetSuite, a major cloud-based service and ERP, will stop allowing any connections made using HMAC-SHA1 after their next release in August 2021 (exact date TBD).  As such, I would like to have the ability to switch to HMAC-SHA256 as a signature method, so there will be no interruptions in any integrations using this library.

### PR DESCRIPTION
**Pull request by svonduhn**
I have added OAuthHmacSha256Signer to enable using the signature method HMAC-SHA256.

Unit test/code coverage has been added in OAuthHmacSha256SignerTest.

This change is necessary for establishing connections to NetSuite as they will no longer accept HMAC-SHA1 as a signature method.

Fixes #702

This replaces pull request #703, which should be closed in favor of this one.  The code is the same (two of the four files modified in the original PR were reverted back to their original versions), so there are only two new files, one class and a test class.

### CODE REVIEW

**Review by TimurSadykov — APPROVED** [review]

File: google-oauth-client/src/main/java/com/google/api/client/auth/oauth/OAuthHmacSha256Signer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.client.auth.oauth;
+
+import com.google.api.client.util.Base64;
+import com.google.api.client.util.Beta;
```
Comment (TimurSadykov) [inline google-oauth-client/src/main/java/com/google/api/client/auth/oauth/OAuthHmacSha256Signer.java:18]: those are deprecated (internally uses com.google.common.io.BaseEncoding), switch to com.google.common.io.BaseEncoding directly to cleanup warnings?

Comment (svonduhn) [reply google-oauth-client/src/main/java/com/google/api/client/auth/oauth/OAuthHmacSha256Signer.java]: This change has been made

**Comment by TimurSadykov** [comment]
@svonduhn you can run "mvn com.coveo:fmt-maven-plugin:format" to fix linter issues

**Comment by svonduhn** [comment]
@TimurSadykov thank you for that command.  Those issues should now be fixed as well.

**Comment by TimurSadykov** [comment]
@svonduhn There is still a linter error, but seems like unrelated to your change, i'm looking into it..

**Review by Neenu1995 — APPROVED** [review]

**Comment by TimurSadykov** [comment]
@svonduhn linter is fixed, you can merge if you are ready
We will release shortly after

**Comment by svonduhn** [comment]
@TimurSadykov Everything seems fine to me.  I don't have access to merge the request, but it looks like it is ready for whoever is able to merge it.

**Comment by TimurSadykov** [comment]
@svonduhn thanks for contributing!

**Pull request merged**